baselines/pFedHN/pFedHN/models.py

```diff
@@ -0,0 +1,222 @@
+            )
+            self.c2_bias = nn.Linear(hidden_dim, 2 * self.n_kernels)
+            self.l1_weights = nn.Linear(hidden_dim, 120 * 2 * self.n_kernels * 5 * 5)
+            self.l1_bias = nn.Linear(hidden_dim, 120)
+            self.l2_weights = nn.Linear(hidden_dim, 84 * 120)
+            self.l2_bias = nn.Linear(hidden_dim, 84)
+            self.l3_weights = nn.Linear(hidden_dim, self.out_dim * 84)
+            self.l3_bias = nn.Linear(hidden_dim, self.out_dim)
+
+        elif self.in_channels == 1:
+            self.c1_weights = nn.Linear(
+                hidden_dim, self.n_kernels * self.in_channels * 3 * 3
+            )
+            self.c1_bias = nn.Linear(hidden_dim, self.n_kernels)
+            self.c2_weights = nn.Linear(
+                hidden_dim, 2 * self.n_kernels * self.n_kernels * 3 * 3
+            )
+            self.c2_bias = nn.Linear(hidden_dim, 2 * self.n_kernels)
+            self.c3_weights = nn.Linear(
+                hidden_dim, 2 * 2 * self.n_kernels * 2 * self.n_kernels * 3 * 3
+            )
+            self.c3_bias = nn.Linear(hidden_dim, 2 * 2 * self.n_kernels)
+            self.l1_weights = nn.Linear(
+                hidden_dim, 120 * 2 * 2 * self.n_kernels * 3 * 3
+            )
+            self.l1_bias = nn.Linear(hidden_dim, 120)
+            self.l2_weights = nn.Linear(hidden_dim, 84 * 120)
+            self.l2_bias = nn.Linear(hidden_dim, 84)
+            self.l3_weights = nn.Linear(hidden_dim, self.out_dim * 84)
+            self.l3_bias = nn.Linear(hidden_dim, self.out_dim)
+
+        if spec_norm and self.in_channels == 3:
+            self.c1_weights = spectral_norm(self.c1_weights)
+            self.c1_bias = spectral_norm(self.c1_bias)
+            self.c2_weights = spectral_norm(self.c2_weights)
+            self.c2_bias = spectral_norm(self.c2_bias)
+            self.l1_weights = spectral_norm(self.l1_weights)
+            self.l1_bias = spectral_norm(self.l1_bias)
+            self.l2_weights = spectral_norm(self.l2_weights)
+            self.l2_bias = spectral_norm(self.l2_bias)
+            self.l3_weights = spectral_norm(self.l3_weights)
+            self.l3_bias = spectral_norm(self.l3_bias)
+
+        elif spec_norm and self.in_channels == 1:
+            self.c1_weights = spectral_norm(self.c1_weights)
+            self.c1_bias = spectral_norm(self.c1_bias)
+            self.c2_weights = spectral_norm(self.c2_weights)
+            self.c2_bias = spectral_norm(self.c2_bias)
+            self.c3_weights = spectral_norm(self.c3_weights)
+            self.c3_bias = spectral_norm(self.c3_bias)
+            self.l1_weights = spectral_norm(self.l1_weights)
+            self.l1_bias = spectral_norm(self.l1_bias)
+            self.l2_weights = spectral_norm(self.l2_weights)
+            self.l2_bias = spectral_norm(self.l2_bias)
+            self.l3_weights = spectral_norm(self.l3_weights)
+            self.l3_bias = spectral_norm(self.l3_bias)
+
+    def forward(self, idx):
```

Comment:
Because this is a hypernetwork, the output is another network. Maybe its useful to expand a bit the description of this `forward()` method? Basically describing what the input `idx` is, and that the outputted network is parameterised given this index.

Reply:
Essentially the `forward()` is returning a `state_dict` that will be used later

Reply:
Sure @jafermarq

Reply:
Added the description